Lib9c/Action/Buy0.cs

```diff
@@ -60,7 +61,9 @@ public override IAccountStateDelta Execute(IActionContext context)
                         GoldCurrencyState.Address);
                 return states.SetState(ShopState.Address, MarkChanged);
             }
-            
+
+            CheckObsolete(2000000, context);
```

Comment:
Why invoke this method after rehearsal?

Reply:
because `CheckObsolete` always pass in rehearsal mode(rehearsal running blockIndex 0)